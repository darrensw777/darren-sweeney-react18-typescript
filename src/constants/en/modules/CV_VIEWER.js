export const CV_VIEWER = {
    DOWNLOAD: "Click here to download a PDF of my CV",
    CONTACT: "Contact me",
    PAGE: "Page",
    OF: "of",
    NEXT: "Next",
    PREVIOUS: "Previous",
};
