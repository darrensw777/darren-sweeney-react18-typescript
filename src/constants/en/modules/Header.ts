export const HeaderCopy = {
    headerLinks: [
        {
            linkTo: '/',
            label: 'Home',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: true,
        },
        {
            linkTo: '/about',
            label: 'About',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/work',
            label: 'Work',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/music',
            label: 'Music',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/videos',
            label: 'Videos',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/gallery',
            label: 'Gallery',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/contact',
            label: 'Contact',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: true,
        },
    ],
};

export default HeaderCopy;
