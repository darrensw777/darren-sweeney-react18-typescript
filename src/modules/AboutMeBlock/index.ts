/* istanbul ignore file */
export { default } from './AboutMeBlock';
