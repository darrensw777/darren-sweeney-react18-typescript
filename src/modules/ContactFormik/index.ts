/* istanbul ignore file */
export { default } from './ContactFormik';
