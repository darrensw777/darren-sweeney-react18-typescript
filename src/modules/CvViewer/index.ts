/* istanbul ignore file */
export { default } from './CvViewer';
