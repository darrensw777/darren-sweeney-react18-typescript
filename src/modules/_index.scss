@import './Header/Header.scss';
@import './Footer/Footer.scss';
@import './PageHeader/page-header.scss';
@import './AboutMeBlock/about-me-block.scss';
@import './CvViewer/cv-viewer.scss';
@import './SoundCloudPlayer/soundcloud-player.scss';
@import './ImagesGallery/images-gallery.scss';
@import './ContactFormik/contact-formik.scss';
@import './InterestLists/interest-lists.scss';