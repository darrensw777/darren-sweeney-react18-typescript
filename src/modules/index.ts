/* istanbul ignore file */
/*
 * Exporting all the components from the components folder,
 * components can be imported as the example below:
 * import { Logo, Avatar } from 'components';
 */

export { default as Header } from './Header';
export { default as Footer } from './Footer';
export { default as PageHeader } from './PageHeader';
export { default as AboutMeBlock } from './AboutMeBlock';
export { default as CvViewer } from './CvViewer';
export { default as SoundCloudPlayer } from './SoundCloudPlayer';
export { default as ImagesGallery } from './ImagesGallery';
export { default as ContactFormik } from './ContactFormik';
export { default as InterestLists } from './InterestLists';
export { default as HomePageBlockContainer } from './HomePageBlockContainer';
