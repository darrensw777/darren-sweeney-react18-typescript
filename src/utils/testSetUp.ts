import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';

const testSetUp = ({ Component }: { Component: ReactElement }) => {
    const { container, getByText, getByRole, asFragment, getAllByRole, getByTestId, getByTitle } = render(Component, {
        wrapper: MemoryRouter,
    });
    return {
        container,
        getByText,
        getByRole,
        asFragment,
        getAllByRole,
        getByTestId,
        getByTitle,
    };
};

export default testSetUp;
