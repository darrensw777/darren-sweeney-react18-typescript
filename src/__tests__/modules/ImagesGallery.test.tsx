import { ImagesGallery } from 'modules';
import testSetUp from 'utils/testSetUp';

const galleryImages = [
    [
        {
            original: '/img/gallery/kay1.webp',
            thumbnail: '/img/gallery/thumbs/kay1_tn.jpg',
        },
        {
            original: '/img/gallery/kay2.webp',
            thumbnail: '/img/gallery/thumbs/kay2_tn.jpg',
        },
        {
            original: '/img/gallery/kay3.webp',
            thumbnail: '/img/gallery/thumbs/kay3_tn.jpg',
        },
        {
            original: '/img/gallery/kay4.webp',
            thumbnail: '/img/gallery/thumbs/kay4_tn.jpg',
        },
    ],
    [
        {
            original: '/img/gallery/misc1.webp',
            thumbnail: '/img/gallery/thumbs/misc1_tn.jpg',
        },
        {
            original: '/img/gallery/misc2.webp',
            thumbnail: '/img/gallery/thumbs/misc2_tn.jpg',
        },
        {
            original: '/img/gallery/misc3.webp',
            thumbnail: '/img/gallery/thumbs/misc3_tn.jpg',
        },
    ],
];

describe('ImagesGallery module', () => {
    let asFragment: any, container: any;
    beforeEach(() => {
        ({ asFragment, container } = testSetUp({
            Component: <ImagesGallery galleryImages={galleryImages} />,
        }));
    });

    test('matches snapshot', () => {
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders the images gallery', () => {
        const imagesGallery = container.getElementsByClassName('images-gallery');
        expect(imagesGallery.length).toBe(1);
    });
});
