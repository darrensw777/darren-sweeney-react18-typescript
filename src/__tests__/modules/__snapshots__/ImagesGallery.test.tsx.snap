// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagesGallery module matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="moduleStyles__ImagesGalleryWrapper-sc-187oehp-5 cBmuKQ"
  >
    <div
      class="images-gallery"
    >
      <div
        aria-live="polite"
        class="image-gallery"
      >
        <div
          class="image-gallery-content bottom"
        >
          <div
            class="image-gallery-slide-wrapper bottom"
          >
            <button
              aria-label="Previous Slide"
              class="image-gallery-icon image-gallery-left-nav"
              type="button"
            >
              <svg
                class="image-gallery-svg"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1"
                viewBox="6 0 12 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="15 18 9 12 15 6"
                />
              </svg>
            </button>
            <button
              aria-label="Next Slide"
              class="image-gallery-icon image-gallery-right-nav"
              type="button"
            >
              <svg
                class="image-gallery-svg"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1"
                viewBox="6 0 12 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="9 18 15 12 9 6"
                />
              </svg>
            </button>
            <div
              class="image-gallery-swipe"
            >
              <div
                class="image-gallery-slides"
              >
                <div
                  aria-label="Go to Slide 1"
                  class="image-gallery-slide  center "
                  role="button"
                  style="display: inherit; transform: translate3d(0%, 0, 0);"
                  tabindex="-1"
                >
                  <img
                    alt=""
                    class="image-gallery-image"
                    height=""
                    loading="eager"
                    sizes=""
                    src="/img/gallery/kay1.webp"
                    srcset=""
                    title=""
                    width=""
                  />
                </div>
                <div
                  aria-label="Go to Slide 2"
                  class="image-gallery-slide  right "
                  role="button"
                  style="display: inherit; transform: translate3d(100%, 0, 0);"
                  tabindex="-1"
                >
                  <img
                    alt=""
                    class="image-gallery-image"
                    height=""
                    loading="eager"
                    sizes=""
                    src="/img/gallery/kay2.webp"
                    srcset=""
                    title=""
                    width=""
                  />
                </div>
                <div
                  aria-label="Go to Slide 3"
                  class="image-gallery-slide  "
                  role="button"
                  style="display: inherit; transform: translate3d(200%, 0, 0);"
                  tabindex="-1"
                >
                  <div
                    style="height: 100%;"
                  />
                </div>
                <div
                  aria-label="Go to Slide 4"
                  class="image-gallery-slide  left "
                  role="button"
                  style="display: inherit; transform: translate3d(-100%, 0, 0);"
                  tabindex="-1"
                >
                  <img
                    alt=""
                    class="image-gallery-image"
                    height=""
                    loading="eager"
                    sizes=""
                    src="/img/gallery/kay4.webp"
                    srcset=""
                    title=""
                    width=""
                  />
                </div>
              </div>
            </div>
            <button
              aria-label="Play or Pause Slideshow"
              class="image-gallery-icon image-gallery-play-button"
              type="button"
            >
              <svg
                class="image-gallery-svg"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="5 3 19 12 5 21 5 3"
                />
              </svg>
            </button>
            <button
              aria-label="Open Fullscreen"
              class="image-gallery-icon image-gallery-fullscreen-button"
              type="button"
            >
              <svg
                class="image-gallery-svg"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 3H5a2 2 0 0 0-2 2v3m18 0V5a2 2 0 0 0-2-2h-3m0 18h3a2 2 0 0 0 2-2v-3M3 16v3a2 2 0 0 0 2 2h3"
                />
              </svg>
            </button>
          </div>
          <div
            class="image-gallery-thumbnails-wrapper bottom thumbnails-swipe-horizontal"
          >
            <div
              class="image-gallery-thumbnails"
            >
              <nav
                aria-label="Thumbnail Navigation"
                class="image-gallery-thumbnails-container"
                style="transform: translate3d(0px, 0, 0); transition: all 450ms ease-out;"
              >
                <button
                  aria-label="Go to Slide 1"
                  aria-pressed="true"
                  class="image-gallery-thumbnail active"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="image-gallery-thumbnail-inner"
                  >
                    <img
                      class="image-gallery-thumbnail-image"
                      src="/img/gallery/thumbs/kay1_tn.jpg"
                    />
                  </span>
                </button>
                <button
                  aria-label="Go to Slide 2"
                  aria-pressed="false"
                  class="image-gallery-thumbnail"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="image-gallery-thumbnail-inner"
                  >
                    <img
                      class="image-gallery-thumbnail-image"
                      src="/img/gallery/thumbs/kay2_tn.jpg"
                    />
                  </span>
                </button>
                <button
                  aria-label="Go to Slide 3"
                  aria-pressed="false"
                  class="image-gallery-thumbnail"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="image-gallery-thumbnail-inner"
                  >
                    <img
                      class="image-gallery-thumbnail-image"
                      src="/img/gallery/thumbs/kay3_tn.jpg"
                    />
                  </span>
                </button>
                <button
                  aria-label="Go to Slide 4"
                  aria-pressed="false"
                  class="image-gallery-thumbnail"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="image-gallery-thumbnail-inner"
                  >
                    <img
                      class="image-gallery-thumbnail-image"
                      src="/img/gallery/thumbs/kay4_tn.jpg"
                    />
                  </span>
                </button>
              </nav>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-live="polite"
        class="image-gallery"
      >
        <div
          class="image-gallery-content bottom"
        >
          <div
            class="image-gallery-slide-wrapper bottom"
          >
            <button
              aria-label="Previous Slide"
              class="image-gallery-icon image-gallery-left-nav"
              type="button"
            >
              <svg
                class="image-gallery-svg"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1"
                viewBox="6 0 12 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="15 18 9 12 15 6"
                />
              </svg>
            </button>
            <button
              aria-label="Next Slide"
              class="image-gallery-icon image-gallery-right-nav"
              type="button"
            >
              <svg
                class="image-gallery-svg"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1"
                viewBox="6 0 12 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="9 18 15 12 9 6"
                />
              </svg>
            </button>
            <div
              class="image-gallery-swipe"
            >
              <div
                class="image-gallery-slides"
              >
                <div
                  aria-label="Go to Slide 1"
                  class="image-gallery-slide  center "
                  role="button"
                  style="display: inherit; transform: translate3d(0%, 0, 0);"
                  tabindex="-1"
                >
                  <img
                    alt=""
                    class="image-gallery-image"
                    height=""
                    loading="eager"
                    sizes=""
                    src="/img/gallery/misc1.webp"
                    srcset=""
                    title=""
                    width=""
                  />
                </div>
                <div
                  aria-label="Go to Slide 2"
                  class="image-gallery-slide  right "
                  role="button"
                  style="display: inherit; transform: translate3d(100%, 0, 0);"
                  tabindex="-1"
                >
                  <img
                    alt=""
                    class="image-gallery-image"
                    height=""
                    loading="eager"
                    sizes=""
                    src="/img/gallery/misc2.webp"
                    srcset=""
                    title=""
                    width=""
                  />
                </div>
                <div
                  aria-label="Go to Slide 3"
                  class="image-gallery-slide  left "
                  role="button"
                  style="display: inherit; transform: translate3d(-100%, 0, 0);"
                  tabindex="-1"
                >
                  <img
                    alt=""
                    class="image-gallery-image"
                    height=""
                    loading="eager"
                    sizes=""
                    src="/img/gallery/misc3.webp"
                    srcset=""
                    title=""
                    width=""
                  />
                </div>
              </div>
            </div>
            <button
              aria-label="Play or Pause Slideshow"
              class="image-gallery-icon image-gallery-play-button"
              type="button"
            >
              <svg
                class="image-gallery-svg"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="5 3 19 12 5 21 5 3"
                />
              </svg>
            </button>
            <button
              aria-label="Open Fullscreen"
              class="image-gallery-icon image-gallery-fullscreen-button"
              type="button"
            >
              <svg
                class="image-gallery-svg"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 3H5a2 2 0 0 0-2 2v3m18 0V5a2 2 0 0 0-2-2h-3m0 18h3a2 2 0 0 0 2-2v-3M3 16v3a2 2 0 0 0 2 2h3"
                />
              </svg>
            </button>
          </div>
          <div
            class="image-gallery-thumbnails-wrapper bottom thumbnails-swipe-horizontal"
          >
            <div
              class="image-gallery-thumbnails"
            >
              <nav
                aria-label="Thumbnail Navigation"
                class="image-gallery-thumbnails-container"
                style="transform: translate3d(0px, 0, 0); transition: all 450ms ease-out;"
              >
                <button
                  aria-label="Go to Slide 1"
                  aria-pressed="true"
                  class="image-gallery-thumbnail active"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="image-gallery-thumbnail-inner"
                  >
                    <img
                      class="image-gallery-thumbnail-image"
                      src="/img/gallery/thumbs/misc1_tn.jpg"
                    />
                  </span>
                </button>
                <button
                  aria-label="Go to Slide 2"
                  aria-pressed="false"
                  class="image-gallery-thumbnail"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="image-gallery-thumbnail-inner"
                  >
                    <img
                      class="image-gallery-thumbnail-image"
                      src="/img/gallery/thumbs/misc2_tn.jpg"
                    />
                  </span>
                </button>
                <button
                  aria-label="Go to Slide 3"
                  aria-pressed="false"
                  class="image-gallery-thumbnail"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="image-gallery-thumbnail-inner"
                  >
                    <img
                      class="image-gallery-thumbnail-image"
                      src="/img/gallery/thumbs/misc3_tn.jpg"
                    />
                  </span>
                </button>
              </nav>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
